src/Console/Output/Progress/NullOutput.php

```diff
@@ -17,6 +17,8 @@
 use PhpCsFixer\Runner\Event\FileProcessed;
 
 /**
+ * @readonly
```

Comment:
TBH, I do not think marking almost all classes `@readonly` is the right step although they are readonly.

The reason is such annotation is only making the code longer. Such check should be done by human review as the human can check 1000x other bad design patterns that the annotation cannot.

Also, if someone will contribute anything, I belive he will not add `@readonly` as this is not common coding practise, thus the code will be only inconsistent then.

Reply:
I have in mind fantastic automation for it , in future ;)

Reply:
I would postpone it until automatized.

Reply:
I don't see that as requirement

Reply:
The final decision is on you, I view it this way: The automation is the one who should add/check these, so the automation should be added, run and verifyed againt this PR. Backwards, veryfying if the already added annotations are correct, is not bullet proof, like merging changes without tests.

Reply:
happy with https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8275#discussion_r1840696276